src/main/java/org/craftercms/engine/graphql/impl/fetchers/ContentTypeBasedDataFetcher.java

```diff
@@ -218,6 +219,15 @@ protected void processSelection(String path, Selection currentSelection, BoolQue
 
             // If the field has sub selection
             if (Objects.nonNull(currentField.getSelectionSet())) {
+                // If the field is a flattened component
+                if (fullPath.matches(COMPONENT_INCLUDE_REGEX)) {
+                    // Include the 'content-type' field to make sure the type can be resolved during runtime
+                    String contentTypeFieldPath = fullPath + "." + QUERY_FIELD_NAME_CONTENT_TYPE;
```

Comment:
Is the content type an XML attribute of the `item` element?

Reply:
It's a child element of `item.component`